src/battle_util.c

```diff
@@ -3481,7 +3481,7 @@ u8 AtkCanceller_UnableToUseMove(void)
         case CANCELLER_FROZEN: // check being frozen
             if (gBattleMons[gBattlerAttacker].status1 & STATUS1_FREEZE && !(gBattleMoves[gCurrentMove].flags & FLAG_THAW_USER))
             {
-                if (Random() % 5)
+                if (RandomWeighted(WRNG_FROZEN, 1, 4))
```

Comment:
Generally I think these changes will make the code a lot more readable, but I'm having a bit of trouble parsing those `RandomWeighted` statements.

Would it be more readable if we wrote
```
if (RandomWeighted(WEIGHTED_RNG_FROZEN, 20, 80))
```
or even had something like `RandomPercent(WEIGHTED_RNG_FROZEN, 80)`?